kong/plugins/ldap-auth/access.lua

```diff
@@ -165,7 +165,7 @@ local function do_authentication(conf)
 
   -- If both headers are missing, return 401
   if not (authorization_value or proxy_authorization_value) then
-    ngx.header["WWW-Authenticate"] = 'LDAP realm="kong"'
+    ngx.header["WWW-Authenticate"] = conf.header_type .. ' realm="kong"'
```

Comment:
existing configurations will mostly have the old default set `ldap` (lowercase) so wouldn't this break an existing configuration after an upgrade?

Adding a migration that converts `ldap` to `LDAP` should fix the defaults. Not sure though about folks not using the default. But probably since no one complained about this issue before, it's safe to just convert `ldap` to `LDAP`?

What do you think @bungle ?

Reply:
I must say I had overlooked this issue. Yet, I'm not sure any client actually does anything with the LDAP type in `WWW-Authenticate`, since it's not listed in the [list maintained by IANA](https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml)

Reply:
I capitalized the default value so as to avoid a breaking change from `WWW-Authenticate: LDAP` to `WWW-Authenticate: ldap` in the response (It doesn't impact the `Authentication: ldap xxxxxxx` header in a valid request since this check is case insensitive). That was not a good option since I overlooked the fact that the default value is stored in the DB and so will only be right in a fresh install.

Here are the options I see:
1. Keep the original default value `ldap`, in which case the default `WWW-Authenticate` header will change to lowercase
2. Change the default value to `LDAP`, and, as @Tieske proposed, add a migration step
3. Change my code to `ngx.header["WWW-Authenticate"] = upper(conf.header_type) .. ' realm="kong"'` but in the case of basic auth, Kong would answer `WWW-Authenticate: BASIC` while the official answer should be `WWW-Authenticate: Basic`

2 seems like the best option... If you agree with that, I can write the migration.

Regarding @Tieske remark:
> Not sure though about folks not using the default.

If someone does not use the default, then the `WWW-Authenticate: LDAP` is not a valid response, so we'll fix a bug for them (actually, that's exactly the issue I have with basic auth).

Reply:
2. is more work but sounds the sanest indeed. Although a migration means that this _must_ target next (which it does) and will only make it into the next major release (far from now).

A possible solution could be to keep the updated default to LDAP:
```lua
local scheme = conf.header_type
if scheme == "ldap" then
  -- ensure backwards compatibility (see GH PR #3656)
  -- TODO: provide migration to capitalize older configurations
  scheme = upper(scheme)
end

ngx.header["WWW-Authenticate"] = scheme .. ' realm="kong"'
```

This way, this PR can go to master. A subsequent PR (that you can contribute later at your own pace) would introduce the migration (there are other similar migrations to copy-paste and tweak, it should be fairly simple), remove this piece code, and target the next branch.